src/terminal.rs

```diff
@@ -0,0 +1,78 @@
+use std::io::{Stdin, Stdout, Write};
+use crate::terminalerror::TerminalError;
+use crate::todo::Todo;
+use console::style;
+
+pub struct Terminal {
+    stdin: Stdin,
+    stdout: Stdout,
+}
+
+impl Terminal {
+    pub fn new() -> Self {
+        Terminal {
+            stdin: std::io::stdin(),
+            stdout: std::io::stdout(),
+        }
+    }
+
+    fn input(&mut self) -> Result<String, TerminalError> {
+        let mut buf: String = String::new();
+
+        match self.stdin.read_line(&mut buf) {
+            Ok(_) => Ok(buf.trim().to_string()),
+            Err(error) => Err(TerminalError::Stdin(error))
+        }
+    }
+    fn should_ask_for_new_todo(&mut self) -> Result<bool, TerminalError> {
+        let mut res = self.input()?;
+        while res.to_lowercase() != "sim" && res.to_lowercase() != "nao" {
+            println!("{}", style("COMANDO ERRADO").red());
+            println!("Digite {} ou {}", style("Sim").blue().bold(), style("Nao").yellow().bold());
+            res = self.input()?;
+        }
+        if res.to_lowercase() == "sim" {
+            Ok(true)
+        } else {
+            Ok(false)
+        }
```

Comment:
Hmmm, essa ruma de `.to_lowercase` aqui é meio estranho, o que vc acha da versão abaixo?

```suggestion
        let mut res = self.input()?.to_lowercase();
        
        while !matches!(&*res, "sim" | "nao") {
            println!("{}", style("COMANDO ERRADO").red());
            println!("Digite {} ou {}", style("Sim").blue().bold(), style("Nao").yellow().bold());
            res = self.input()?.to_lowercase();
        }
        
        Ok(res == "sim")
```